core/src/main/java/org/apache/iceberg/ManifestListWriter.java

```diff
@@ -31,14 +31,18 @@
 abstract class ManifestListWriter implements FileAppender<ManifestFile> {
   private final FileAppender<ManifestFile> writer;
 
-  private ManifestListWriter(OutputFile file, Map<String, String> meta) {
-    this.writer = newAppender(file, meta);
+  private ManifestListWriter(
+      OutputFile file,
+      Map<String, String> meta,
+      String compressionCodec,
+      Integer compressionLevel) {
+    this.writer = newAppender(file, meta, compressionCodec, compressionLevel);
   }
 
   protected abstract ManifestFile prepare(ManifestFile manifest);
 
   protected abstract FileAppender<ManifestFile> newAppender(
-      OutputFile file, Map<String, String> meta);
+      OutputFile file, Map<String, String> meta, String compressionCodec, Integer compressionLevel);
```

Comment:
this is what's causing the RevAPI breaks. I think it would be better to have 
```
protected FileAppender<ManifestFile> newAppender(
      OutputFile file,
      Map<String, String> meta,
      String compressionCodec,
      Integer compressionLevel) {
    return newAppender(file, meta, null, null);
  }
```
which calls the original `newAppender(..)` method with either nulls or default values for compression codec & level

Reply:
This is not what's causing the revapi breakage; it is the `newAppender` in `ManifestWriter`.
`ManifestListWriter` is package private, not public, so the change is not an API break.

Reply:
you're right, it's actually `ManifestWriter` and not `ManifestListWriter` (rev api breakages are sometimes difficult to read). 
As mentioned further above we shouldn't be breaking the API (yet) as we need to go through one Deprecation cycle as indicated in https://github.com/apache/iceberg/blob/master/CONTRIBUTING.md#minor-version-deprecations-required